### ISSUE TITLE
AstroidBuildingException with weird but valid syntax

### ISSUE THREAD

**Issue by Juanlu001** [PyCQA/pylint#910]
### Steps to reproduce
pylint fails with the code below:

```
> type test.py
import collections
collections.namedtuple('{}'.format("a"), '')()
```

### Current behavior

```
> pylint test.py
No config file found, using default configuration
************* Module config
C:  1, 0: Missing module docstring (missing-docstring)
Traceback (most recent call last):
  File "c:\python27\lib\runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "c:\python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\Scripts\pylint.exe\__main__.py", line 9, in <module>
  File "c:\python27\lib\site-packages\pylint\__init__.py", line 23, in run_pylint
    Run(sys.argv[1:])
  File "c:\python27\lib\site-packages\pylint\lint.py", line 1315, in __init__
    linter.check(args)
  File "c:\python27\lib\site-packages\pylint\lint.py", line 736, in check
    self._do_check(files_or_modules)
  File "c:\python27\lib\site-packages\pylint\lint.py", line 867, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "c:\python27\lib\site-packages\pylint\lint.py", line 947, in check_astroid_module
    walker.walk(ast_node)
  File "c:\python27\lib\site-packages\pylint\utils.py", line 938, in walk
    self.walk(child)
  File "c:\python27\lib\site-packages\pylint\utils.py", line 938, in walk
    self.walk(child)
  File "c:\python27\lib\site-packages\pylint\utils.py", line 935, in walk
    cb(astroid)
  File "c:\python27\lib\site-packages\pylint\checkers\stdlib.py", line 186, in visit_call
    for inferred in node.func.infer():
  File "c:\python27\lib\site-packages\astroid\bases.py", line 364, in infer
    return self._explicit_inference(self, context, **kwargs)
  File "c:\python27\lib\site-packages\astroid\brain\brain_stdlib.py", line 303, in infer_named_tuple
    ''' % {'name': name, 'fields': attributes})
  File "c:\python27\lib\site-packages\astroid\builder.py", line 146, in string_build
    module = self._data_build(data, modname, path)
  File "c:\python27\lib\site-packages\astroid\builder.py", line 174, in _data_build
    raise exceptions.AstroidBuildingException(exc)
astroid.exceptions.AstroidBuildingException: invalid syntax (<string>, line 2)
```

### Expected behavior

At least, finish the analysis until the end of the file.

### pylint --version output

```
No config file found, using default configuration
pylint 1.5.5,
astroid 1.4.5
Python 2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:40:30) [MSC v.1500 64 bit (AMD64)]
```

**Comment by PCManticore** [comment]
This should be fixed now on the latest astroid (https://github.com/PyCQA/astroid/commit/e888fb3c45178929259b8544100c5b9e70db7a79) I'll try to make a short bug fix release for astroid in the weekend.

**Comment by Juanlu001** [comment]
That's great, thanks for your quick response!

**Comment by Juanlu001** [comment]
Hi @PCManticore, any ETA for astroid 1.4.6? I looked for a milestone in the repository for an estimate but found none. No pressure, I just want to know if it will be days, weeks or months :)

cc @skasi7

**Comment by PCManticore** [comment]
Thank you for reminding me, I totally forgot about it. I can do it later today. In the eventuality I won't do it, please ping me again.

**Comment by Juanlu001** [comment]
No worries, I'll comment here again tomorrow in case I don't see a new astroid release =) Thanks!

**Comment by PCManticore** [comment]
Just released astroid 1.4.6 and pylint 1.5.6. Let me know how it works for you!

**Comment by Juanlu001** [comment]
I just checked and it works flawlessly :smile: Thanks again!
